client/app/pages/queries/view.js

```diff
@@ -204,6 +204,25 @@ function QueryViewCtrl($scope, Events, $route, $routeParams, $location, $window,
     $scope.saveQuery(undefined, { is_draft: $scope.query.is_draft });
   };
 
+  $scope.saveAsDataset = () => {
```

Comment:
No need for additional dependencies, the new syntax is native to javascript and requires no transpiler.